### PR TITLE
feat(Trading reward):  Top Traders

### PR DESCRIPTION
**Pull request by ChefMomota**
### Summary
🏆🍰📊

This pull request adds new features and components for the trading reward page on the web app. It introduces a new custom hook `useContract` to access the `tradingRewardTopTrades` contract, which tracks and rewards the top traders on PancakeSwap. It also adds a new page `trading-reward/top-traders.tsx` that shows the leaderboard of the top traders, and a new sub menu component `SubMenu` that allows users to navigate between the trading reward pages. It also modifies some existing components and constants to support the new features and improve the layout of the trading reward page.

> _`tradingRewardTopTrades`_
> _Rewards the best traders_
> _A new contract_

### Walkthrough
*  Add a new property `hideSubNav` to the menu config for the trading reward page to hide the sub navigation bar ([link](https://github.com/pancakeswap/pancake-frontend/pull/7059/files?diff=unified&w=0#diff-e0daeed87867fc4fc6b0b1a9c2d52c9bfbd175cd1822e100f43300c4046484adR117))
*  Modify the import and usage of the `useAllTradingRewardPair` hook to accept an object with `status` and `type` parameters for filtering the trading reward pairs ([link](https://github.com/pancakeswap/pancake-frontend/pull/7059/files?diff=unified&w=0#diff-188a4fc41bafbe3196976f16f7309d2152bb8ed76ba789dcb87f289057eeff14L2-R9), [link](https://github.com/pancakeswap/pancake-frontend/pull/7059/files?diff=unified&w=0#diff-750bc10b5a40d71eefc9da39629f7004b51d41beeefac6b05b0b5d25937e76c2L5-R12))
*  Add a new contract address for the `tradingRewardTopTrades` contract on the BSC mainnet and a new helper function to get the contract address ([link](https://github.com/pancakeswap/pancake-frontend/pull/7059/files?diff=unified&w=0#diff-6ccab19ffbec70686dc2221f7b79432ece4302c36067c91063d258209c09ec07R201-R205), [link](https://github.com/pancakeswap/pancake-frontend/pull/7059/files?diff=unified&w=0#diff-05729caa7bf80ee8f359513f222175a79f3cd3e8aa704d68f0d5c67f8b010a23R130-R133))
*  Modify the `TRADING_REWARD_API` constant to use a different URL for the trading reward API that supports the new features ([link](https://github.com/pancakeswap/pancake-frontend/pull/7059/files?diff=unified&w=0#diff-7238357db46da6bb48ca41841457ac0e832ee21fb8fd76a6b3c9dbf701ce8c62L60-R60))
*  Add a new helper function and a new custom hook to get the contract instance for the `tradingRewardTopTrades` contract ([link](https://github.com/pancakeswap/pancake-frontend/pull/7059/files?diff=unified&w=0#diff-0189a20b061b4c4cd5fd74abb3ff0a22a05b9f5adde9583eb11606cb589a5af4R46), [link](https://github.com/pancakeswap/pancake-frontend/pull/7059/files?diff=unified&w=0#diff-0189a20b061b4c4cd5fd74abb3ff0a22a05b9f5adde9583eb11606cb589a5af4R384-R389), [link](https://github.com/pancakeswap/pancake-frontend/pull/7059/files?diff=unified&w=0#diff-dbcbcd450dce97f74ff9b8fff28d36fc457775363d349db2a45e504060c0754aR33), [link](https://github.com/pancakeswap/pancake-frontend/pull/7059/files?diff=unified&w=0#diff-dbcbcd450dce97f74ff9b8fff28d36fc457775363d349db2a45e504060c0754aR371-R379))
*  Add a new file for the trading reward page that exports the `TradingReward` component as the default component ([link](https://github.com/pancakeswap/pancake-frontend/pull/7059/files?diff=unified&w=0#diff-ff74f35286f5b8ba5ae325c65cc8e5fa463231fcfc1c8668e7a289f7cf7b5aeaR1-R3))
*  Add a new file for the trading reward top traders page that exports the `TradingRewardTopTraders` component as the default component ([link](https://github.com/pancakeswap/pancake-frontend/pull/7059/files?diff=unified&w=0#diff-cbe2fec37ed128615a6d5de3cf02380056a0d0db44fb3967d104fffe78fa49a1R1-R3))
*  Add a new component `SubMenu` that renders the sub menu items for the trading reward page using the uikit and the router hooks ([link](https://github.com/pancakeswap/pancake-frontend/pull/7059/files?diff=unified&w=0#diff-853e95df69c09f4098199ad34def4085a94e5376e2ac7479f7c5245c8b7ea598R1-R24))
*  Add a new component `Leaderboard` that renders the leaderboard section for the trading reward page using various hooks and subcomponents ([link](https://github.com/pancakeswap/pancake-frontend/pull/7059/files?diff=unified&w=0#diff-94382753793192a4edd427aba2521b380c88958dbcc9821b7f456a923c67847aR1-R100))
*  Add new components `DesktopView`, `DesktopResult`, `MobileView`, `MobileResult`, and `RankingCard` that render the leaderboard for different screen sizes and ranks using various hooks and utils ([link](https://github.com/pancakeswap/pancake-frontend/pull/7059/files?diff=unified&w=0#diff-4b8800b925873b06295e95f7b8ab1cdb221b8d7898355d43b84d79c9116fa730R1-R53), [link](https://github.com/pancakeswap/pancake-frontend/pull/7059/files?diff=unified&w=0#diff-5fd07b53fd6d3e6d1762ae5cd1d0b3ac32bdef19353c7d95340934d8359cf130R1-R65), [link](https://github.com/pancakeswap/pancake-frontend/pull/7059/files?diff=unified&w=0#diff-4bf926d7d5e072659dfcf6f3db3ea66aa5958caa1849f1b86096c094fcc95d16R1-R75), [link](https://github.com/pancakeswap/pancake-frontend/pull/7059/files?diff=unified&w=0#diff-f220411cac4174fd35a3d24356555fc08bb11f1ab78349cdceabc7f55ee45d5aR1-R53), [link](https://github.com/pancakeswap/pancake-frontend/pull/7059/files?diff=unified&w=0#diff-1fad17216e1fcaa96fd7f6dbc9fd991da236bfa3731d787ed343490d91bdcbe8R1-R109))
*  Modify the `RewardsBreakdown` component to increase the max-height of the card to show more rows of the table ([link](https://github.com/pancakeswap/pancake-frontend/pull/7059/files?diff=unified&w=0#diff-83004c4692a68e9f5776d408fb1986e774233934f3d97b19761f7d3fe6d3fa01L95-R95))
*  Modify the `Questions` component to increase the max-width of the wrapper to make the layout more consistent ([link](https://github.com/pancakeswap/pancake-frontend/pull/7059/files?diff=unified&w=0#diff-87aa04a00caf6799977d3195e92fc044abb284ba1b495fd9ae96447844485e43L19-R19))
*  Delete the file `apps/web/src/pages/trading-reward.tsx` ([link](https://github.com/pancakeswap/pancake-frontend/pull/7059/files?diff=unified&w=0#diff-101167700b8fa92c19c8f009e5e17d6c9fd629f94d7e8066873013497d4ec164))

### CODE REVIEW

File: apps/web/src/config/constants/endpoints.ts [hunk 1]
```diff
@@ -57,7 +57,7 @@ export const V3_SUBGRAPH_URLS = {
   [ChainId.BSC_TESTNET]: 'https://api.thegraph.com/subgraphs/name/pancakeswap/exchange-v3-chapel',
 } satisfies Record<ChainId, string>
 
-export const TRADING_REWARD_API = 'https://pancake-trading-fee-rebate-api.pancake.run/api/v1'
+export const TRADING_REWARD_API = 'https://pancake-trading-fee-rebate-api-git-feat-dev.pancake.run/api/v1'
```
Comment (ChefMomota) [inline apps/web/src/config/constants/endpoints.ts:60]: revert it before deploy

File: apps/web/src/views/TradingReward/components/Leaderboard/DesktopView.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+            </>
+          )}
+        </tbody>
+      </Table>
+      <PaginationButton showMaxPageText currentPage={currentPage} maxPage={maxPage} setCurrentPage={setCurrentPage} />
+    </Card>
+  )
+}
+
+export default DesktopView
```
Comment (chefjackson) [inline apps/web/src/views/TradingReward/components/Leaderboard/DesktopView.tsx:65]: Let's use named exports if possible

File: apps/web/src/views/TradingReward/components/TopTraders/CurrentRewardPool.tsx [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+        <Flex justifyContent="space-between" mb="10px">
+          <Text color="white" fontWeight={['400', '400', '400', '600']} fontSize={['14px', '14px', '14px', '20px']}>
+            {t('Total volume generated')}
+          </Text>
+          <Text bold color="white" fontSize={['14px', '14px', '14px', '20px']}>
+            {`$${formatNumber(campaignInfoData?.totalVolume, 3, 3)}`}
+          </Text>
+        </Flex>
+        <Flex justifyContent="space-between" mb="10px">
+          <Text color="white" fontWeight={['400', '400', '400', '600']} fontSize={['14px', '14px', '14px', '20px']}>
```
Comment (chefjackson) [inline apps/web/src/views/TradingReward/components/TopTraders/CurrentRewardPool.tsx:160]: The `Text` component looks repetitive. Can we refactor into using one component so we don't need to define the same props all over the place?

File: apps/web/src/views/TradingReward/components/TopTraders/YourTradingReward/CurrentPeriod.tsx [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+                            <Text bold fontSize="14px" color="primary" as="span" ml="4px">
+                              {`${timeUntil.days}${t('d')}`}
+                            </Text>
+                          ) : null}
+                          {timeUntil.days || timeUntil.hours ? (
+                            <Text bold fontSize="14px" color="primary" as="span" ml="4px">
+                              {`${timeUntil.hours}${t('h')}`}
+                            </Text>
+                          ) : null}
+                          <Text bold fontSize="14px" color="primary" as="span" ml="4px">
```
Comment (chefjackson) [inline apps/web/src/views/TradingReward/components/TopTraders/YourTradingReward/CurrentPeriod.tsx:141]: Same here. Same props is defined multiple times

File: apps/web/src/views/TradingReward/hooks/useRankList.tsx [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+    campaignId && currentPage && ['/trader-rank-list', campaignId, currentPage],
+    async () => {
+      try {
+        if (currentPage * MAX_PER_PAGE > topTradersArr.length) {
+          setIsLoading(true)
+          const response = await fetch(
+            `${TRADING_REWARD_API}/rank_list/campaignId/${campaignId}/type/tt/page/${currentPage}/size/${MAX_PER_PAGE}`,
+          )
+          const result: RankListResponse = await response.json()
+          setTotal(result.data.total)
```
[the remark this answers is not in the document]
Comment (ChefMomota) [reply apps/web/src/views/TradingReward/hooks/useRankList.tsx]: done

File: apps/web/src/config/constants/contracts.ts [hunk 6]
```diff
@@ -176,7 +176,7 @@ export default {
   },
   tradingReward: {
     1: '0x',
-    56: '0xa842a4AD40FEbbd034fbeF25C7a880464a90e695',
+    56: '0xe9f80dfC6ddfc4637b8B3BdFc6838209Dc82eF95',
```
Comment (ChefMomota) [inline apps/web/src/config/constants/contracts.ts:179]: revert it before deploy

**Pull request merged**